cypress/e2e/models/administration/jira/jira.ts

```diff
@@ -185,26 +194,36 @@ export class Jira {
         performRowAction(this.name, deleteAction);
         if (toBeCanceled) {
             click(cancelButton);
-            exists(this.name);
+            this.validateState();
         } else {
             click(confirmButton);
-            notExists(this.name);
+            notExists(this.name, "table[aria-label='Jira trackers table']");
         }
     }
 
     /**
      * This method validates all fields values of Jira connection after creation
      */
-    public validateState(): void {
+    public validateState(expectedToFail = false): void {
+        let expectedStatus: string;
+        if (expectedToFail) {
+            expectedStatus = "Not connected";
+        } else {
+            expectedStatus = "Connected";
+        }
+        // This is horrible, but somehow further code inside `cy.get()` takes OLD values from the object and I need to define separate values and use them to override this problem.
+        let name = this.name;
+        let url = this.url;
```

Comment:
**nitpick:** I understand that inside the `within` the closure is different so it can't access the `this` jira instance, so the comment is not really needed.
 
 On the other hand, if `name` and `url` aren't reassigned, they can be declared as `const`.

Reply:
Changed